File: gitdot-api/src/endpoint/question/list_questions.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{common::Page, question::QuestionResource},
};

pub struct ListQuestions;

impl Endpoint for ListQuestions {
    const PATH: &'static str = "/repository/{owner}/{repo}/questions";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListQuestionsRequest;
    type Response = ListQuestionsResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListQuestionsRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListQuestionsResponse = Page<QuestionResource>;

File: gitdot-web/packages/gitdot-api/src/endpoint/question/list-questions.ts
import { z } from "zod";
import { page, QuestionResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListQuestionsRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListQuestionsRequest = z.infer<typeof ListQuestionsRequest>;

export const ListQuestionsResponse = page(QuestionResource);
export type ListQuestionsResponse = z.infer<typeof ListQuestionsResponse>;

export const ListQuestions = {
  path: "/repository/{owner}/{repo}/questions",
  method: "GET",
  request: ListQuestionsRequest,
  response: ListQuestionsResponse,
} as const satisfies Endpoint;
export type ListQuestions = typeof ListQuestions;
